### PR TITLE
useMove + pokemon affix localization

### PR DESCRIPTION
**Pull request by laeticiapierre**
Issue #1274

EDIT 05/27: 
ENG
Capture d’écran 2024-05-27 à 17 13 48
Capture d’écran 2024-05-27 à 17 13 54

FR
Capture d’écran 2024-05-27 à 17 12 53
Capture d’écran 2024-05-27 à 17 13 01

### CODE REVIEW

**Review by Greenlamp2 — APPROVED** [review]

**Comment by CodeTappert** [comment]
Could you please make it so that both of them could be suffixes as well?

German for example uses: "Glumanda (Wild)" instead of "Wildes Glumanda" (Wild is the "prefix").

Same with the foe one

**Comment by laeticiapierre** [comment]
> Could you please make it so that both of them could be suffixes as well?
> 
> German for example uses: "Glumanda (Wild)" instead of "Wildes Glumanda" (Wild is the "prefix").
> 
> Same with the foe one

I've put `pokemonPrefix` as a parameter name because it's the method name but you can place it wherever you like, feel free to check the french file, it's used that way

**Comment by Dakurei** [comment]
As it's not at the beginning of a sentence in French, it's a suffix rather than a prefix, so there's no need to capitalize it.
Besides being ugly, it's incorrect 😅

**Comment by CodeTappert** [comment]
> > Could you please make it so that both of them could be suffixes as well?
> > German for example uses: "Glumanda (Wild)" instead of "Wildes Glumanda" (Wild is the "prefix").
> > Same with the foe one
> 
> I've put `pokemonPrefix` as a parameter name because it's the method name but you can place it wherever you like, feel free to check the french file, it's used that way

Yeah i saw that. So pokemonPrefix is either wildPrefix or foePrefix? Am i understanding this correctly?

**Comment by laeticiapierre** [comment]
> As it's not at the beginning of a sentence in French, it's a suffix rather than a prefix, so there's no need to capitalize it. Besides being ugly, it's incorrect 😅

We could easily fix this by updating the french file and replacing `Sauvage` with `sauvage`(same for `Ennemi`) but traduction is not done everywhere at the moment, there are some places left where we would have "sauvage Tortank [...]" as a sentence and I think it's better to have a capitalized letter at the start of it

**Comment by CodeTappert** [comment]
German translation. Source (Pokemon Scarlet)
```
  "wildPokemonPrefix": "(wild)",
  "foePokemonPrefix": "(Gegner)",
  "useMove": "{{pokemonName}} {{pokemonPrefix}} setzt {{moveName}} ein!"
```

**Comment by CodeTappert** [comment]
> German translation. Source (Pokemon Scarlet)
> 
> ```
>   "wildPokemonPrefix": " (wild)",
>   "foePokemonPrefix": " (Gegner)",
>   "useMove": "{{pokemonName}}{{pokemonPrefix}} setzt {{moveName}} ein!"
> ```

Just quick update. I updated this again just now so it works even when the prefix is set to an empty string (which i assume is done for our own Pokemon?). 

Just added a space in the prefix itself but removed the one in the useMove string

**Comment by wuzhaoqi1015** [comment]
Has this issue been merged?

**Comment by CodeTappert** [comment]
> Has this issue been merged?

Not yet. The creator of the issue has not adressed the latest comments or made changes suggestet. It is a "Draft". So we wait

**Comment by CodeTappert** [comment]
Could you please provide screenshots of the german version? Would love to see if it looks correct

**Comment by laeticiapierre** [comment]
> Could you please provide screenshots of the german version? Would love to see if it looks correct

Capture d’écran 2024-05-28 à 12 56 10
Capture d’écran 2024-05-28 à 12 56 58

**Comment by CodeTappert** [comment]
> > Could you please provide screenshots of the german version? Would love to see if it looks correct
> 
> Capture d’écran 2024-05-28 à 12 56 10
> Capture d’écran 2024-05-28 à 12 56 58
> 
> 

Thank. You. Looks very good!

**Comment by molker** [comment]
Sorry this is a product of the delay in further review but since korean is now in locales can you please add that to this PR

File: src/messages.ts [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+import i18next from "./plugins/i18n";
 
 export function getPokemonMessage(pokemon: Pokemon, content: string): string {
-  return `${getPokemonPrefix(pokemon)}${pokemon.name}${content}`;
+  return `${getPokemonNameWithAffix(pokemon)} ${content}`;
 }
 
-export function getPokemonPrefix(pokemon: Pokemon): string {
-  let prefix: string;
+export function getPokemonNameWithAffix(pokemon: Pokemon): string {
```
Comment (molker) [inline src/messages.ts:9]: please add documentation for this and the above method, reference `docs/comments.md` for a guide

**Review by molker — CHANGES_REQUESTED** [review]
Looks good, a few things to fix before we can approve

File: src/messages.ts (hunk 1 above)
Comment (laeticiapierre) [reply src/messages.ts:21]: Done 👍🏻 (and korean)

File: src/locales/ko/battle.ts [hunk 2]
```diff
@@ -55,5 +55,8 @@ export const battle: SimpleTranslationEntries = {
   "notDisabled": "{{pokemonName}}의\n{{moveName}} 사슬묶기가 풀렸다!",
   "skipItemQuestion": "아이템을 받지 않고 넘어가시겠습니까?",
   "eggHatching": "어라…?",
-  "ivScannerUseQuestion": "{{pokemonName}}에게 개체값탐지기를 사용하시겠습니까?"
+  "ivScannerUseQuestion": "{{pokemonName}}에게 개체값탐지기를 사용하시겠습니까?",
+  "wildPokemonWithAffix": "Wild {{pokemonName}}",
+  "foePokemonWithAffix": "Foe {{pokemonName}}",
+  "useMove": "{{pokemonNameWithAffix}} used {{moveName}}!"
```
Comment (returntoice) [inline src/locales/ko/battle.ts:61]:
```suggestion
  "wildPokemonWithAffix": "야생 {{pokemonName}}",
  "foePokemonWithAffix": "상대 {{pokemonName}}",
  "useMove": "{{pokemonNameWithAffix}}의 {{moveName}}!"
```
translated by the official translation(VIII, IX gen)

**Review by returntoice — CHANGES_REQUESTED** [review]

**Review by returntoice — APPROVED** [review]

**Comment by flx-sta** [comment]
@laeticiapierre please resolve the merge conflicts

**Review by bennybroseph — APPROVED** [review]

**Pull request merged**